File: longest substring without repeating characters/longest_substring.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        initial_index = 0
        char_index = {}
        max_length = 0
        for i in range(len(s)):
            current_char = s[i]
            if current_char in char_index and char_index[current_char] >= initial_index:
                initial_index = char_index[current_char] + 1
            char_index[current_char] = i
            max_length = max(max_length, i - initial_index + 1)
        return max_length

if __name__ == "__main__":
    sol = Solution()
    result = sol.lengthOfLongestSubstring("aaaaaab")
    print(result)

File: longest substring without repeating characters/longestSubstring.js
var lengthOfLongestSubstring = function(string) {
    let initial_index = 0;
    let maxLength = 0;
    const charIndex = {};
    for (let i = 0; i < string.length; i++) {
        const current_char = string[i];
        if (current_char in charIndex && charIndex[current_char] >= initial_index) {
            initial_index = charIndex[current_char] + 1;
        }
        charIndex[current_char] = i;
        maxLength = Math.max(maxLength, i - initial_index + 1)
    }
    return maxLength
}

const result = lengthOfLongestSubstring("aaaaa");
console.log(result)